automation_tools/scripts/ga-migration/utils.py

```diff
@@ -196,7 +196,19 @@ def replace_regex(regex, output, filepath):
         # TODO: expose number of matches
         with fileinput.FileInput(filepath, inplace=True, backup=".bak") as file:
             for line in file:
-                print(re.sub(regex, output, line), end="")
+                if replace_group == 0:
+                    print(re.sub(regex, output, line), end="")
+                else:
+                    # Match regex in line
+                    m = re.search(regex, line)
+                    if m:
+                        # Get the requested group
+                        matched_group = m.group(replace_group)
+                        # Replace the matched group with the given text
+                        print(re.sub(matched_group, output, line), end="")
+                    else:
+                        print(re.sub(regex, output, line), end="")
+
```

Comment:
What about something like this (untested)?
```suggestion
                pattern = regex
                if replace_group != 0:
                    # Match regex in line
                    m = re.search(regex, line)
                    if m:
                        # Get the requested group
                        pattern = m.group(replace_group)
                print(re.sub(pattern, output, line), end="")
```

Just curious, why do you need to print it?

Reply:
@mvidalgarcia it was the easier way to "skip" the line and re-write it as the same (since we are "consuming" the file line by line. If you do that, you are skipping non-matching lines (thus removing them)